content/en/docs/concepts/workloads/controllers/daemonset.md

```diff
@@ -201,6 +201,63 @@ them according to its `updateStrategy`.
 
 You can [perform a rolling update](/docs/tasks/manage-daemon/update-daemon-set/) on a DaemonSet.
 
+While working with the rolling updates of the DaemonSets, you will come across two important parameters: `maxSurge` and `maxUnavailable`.
+
+### maxSurge
+
+If you set this parameter to a number above 0, you allow the DaemonSet controller
+to create temporary duplicate Pods during a rollout.
+Because a DaemonSet aims to have one healthy Pod per node, this field also implies
+the number of _nodes_ that can have a duplicate Pod on them at any one moment
+during the rollout.
+While performing a rolling update, Kubernetes will keep creating new pods to replace the old ones. 
+`maxSurge` helps you to regulate the surge of additional pods beyond the desired count. 
+
+#### maxSurge and Pod failures
+
+During a DaemonSet rollout, one of the old pods maybe become unavailable for some reason
+(for example: the `Ready` condition transitions to false, the  Pod is evicted, or the node is drained).
+If that occurs, the DaemonSet controller immediately creates a replacement Pod on that node,
+without considering surge limits. The replacement Pod uses the current (new) Pod template, not
+the previous one.
+
+### maxUnavailable
+
+This parameter defines the maximum number of pods, belonging to a DaemonSet,
+that can be unavailable while performing an update.
+In simpler terms, they are being deleted and not yet replaced by new pods, during the update process. 
```

Comment:
hm, I think this can be a bit confusing. There can be a slight delay between the deletion and replacement, but practically it should be unnoticeable. I think it is safe to say that the pods are replaced immediately, but they might take time to become available. that is app becomes `ready` + `minReadySeconds`

Reply:
It depends on how you define replacement:
1. is it just the Pod API object, even if it is no picked up by scheduler yet?
2. is it Pod after being assigned and running?
3. is it available Pod?

If the user expects 1., then it could be confusing. I think it would be better to describe in more detail to prevent the confusion.

```suggestion
During the upgrade process, the pods are being deleted, and it may take some time for the replacement pods to become available on each node.
```

or something similar in this spirit?